languages/typescript/src/main/kotlin/io/vrap/codegen/languages/typescript/joi/JoiValidatorModuleRenderer.kt

```diff
@@ -57,55 +57,71 @@ class JoiValidatorModuleRenderer @Inject constructor(override val vrapTypeProvid
         }
     }
 
-    fun UnionType.renderJoiUnionSchema():String {
+    private fun UnionType.renderJoiUnionSchema():String {
         val arrayArg:String = this.getOneOf().map { "${it.toVrapType().simpleJoiName()}()"}
                 .joinToString(prefix = "[",separator = ",",postfix = "]")
         val schemaDeclaration = """
-            |schema.${toVrapType().simpleJoiName()} = Joi.lazy(() =\> Joi.alternatives($arrayArg))
+            |Joi.alternatives($arrayArg)
             |
         """
         return toVrapType().renderSchemaExportFunction(schemaDeclaration).trimMargin()
     }
 
-    fun VrapType.renderSchemaExportFunction(schemaDeclaration: String): String {
+    private fun VrapType.renderSchemaExportFunction(schemaDeclaration: String): String {
+        val attachedId = if(recursionTypes.contains(simpleJoiName())) ".id('${simpleJoiName()}')" else ""
         return """
-            |export function ${simpleJoiName()}() : Joi.AnySchema {
-            |   if (!schema.${simpleJoiName()}) {
-            |     $schemaDeclaration
-            |   }
-            |   return schema.${simpleJoiName()}
-            |}
+            |export const ${simpleJoiName()} = () =\> ${schemaDeclaration}${attachedId}
```

Comment:
use arrow function instead of normal function cause its lighter (it doesnt have `this` scope)